### ISSUE TITLE
Give TestCase a default testMethod

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2147]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2147|
|Type|enhancement|
|Created|2006-10-03 06:03:27Z|

So, I think there was a bug to remove this. 

I've changed my mind.

 * Add an empty runTest() method to `TestCase`
 * Make 'runTest' the default methodName

<details><summary>Searchable metadata</summary>

```
trac-id__2147 2147
type__enhancement enhancement
reporter__jml jml
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1159855407000000 1159855407000000
changetime__1164860549000000 1164860549000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

What made you change your mind?

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml commented|
|-|-|

Mostly looking at nit.  With the current API, it's not obvious how to make a single testcase. Also there's a stack of tests in Trial that pick arbitrary methods just to construct a test case

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml removed owner|
|-|-|

Ready for review in default-method-2147

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Perhaps the motivations for this should be documented in the code somewhere?

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml commented|
|-|-|

Added. I assume that wasn't a proper review.

What I wanted to say was that a `TestCase` is a complicated and meaningful thing, even if it doesn't have any actual test inside it, that it seems sensible for `TestCase()` to construct a test case that does nothing.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to dash|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_7"></a>|@jerub set owner to @jml|
|-|-|

Excellent work. Everything is clearly docstringed.

Again whitespace changes frustrate my diff, so I've had to use this invokation of svn diff:
```
$  svn diff --diff-cmd diff --extensions '-ubB'
```

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_8"></a>|@jml set status to `closed`|
|-|-|

(In [18791]) Allow `TestCase`s to be constructed without parameters.
 * Author: jml
 * Reviewer: jerub
 * Fixes [#2147](https://github.com/twisted/twisted/issues/2147)

Provides a default test method `runTest` that is used as the test method
when `TestCase` is constructed without parameters. This means it is now
really easy to get the trivial test case.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
